services.py

```
@@ -0,0 +1,116 @@
+        data_json = json.dumps(data)
+        self.element_internal_op(widget_id, data_json)
+
+    def append_data(self, widget_id: int, data: any):
+        data_json = json.dumps(data)
+        self.element_internal_op(widget_id, data_json)
+
+    def reset_data(self, widget_id: int):
+        data_json = json.dumps("")
+        self.element_internal_op(widget_id, data_json)
+
+    def reset_data(self, widget_id: int, data: any):
```

Ruff (0.8.2)

Redefinition of unused `reset_data` from line 77

(F811)